chainercv/datasets/kitti/kitti_bbox_dataset.py

```diff
@@ -0,0 +1,219 @@
+import os
+import warnings
+
+from pkg_resources import get_distribution
+from pkg_resources import parse_version
+
+import numpy as np
+try:
+    import pykitti
+    pykitti_version = get_distribution('pykitti').version
+    if parse_version(pykitti_version) >= parse_version('0.3.0'):
+        # pykitti>=0.3.0
+        _available = True
+    else:
+        # pykitti<0.3.0
+        warnings.warn('not support pykitti version : ' + pykitti_version)
+        _available = False
+except ImportError:
+    _available = False
+
+from chainercv.chainer_experimental.datasets.sliceable import GetterDataset
+from chainercv.datasets.kitti.kitti_utils import get_kitti_label
+from chainercv.datasets.kitti.kitti_utils import get_kitti_nosync_data
+from chainercv.datasets.kitti.kitti_utils import get_kitti_sync_data
+from chainercv.datasets.kitti.kitti_utils import get_kitti_tracklets
+from chainercv.datasets.kitti.kitti_utils import kitti_date_lists
+from chainercv.datasets.kitti.kitti_utils import kitti_date_num_dicts
+
+
+def _check_available():
+    if not _available:
+        raise ValueError(
+            'pykitti is not installed in your environment,'
+            'so the dataset cannot be loaded.'
+            'Please install pykitti to load dataset.\n\n'
+            '$ pip install pykitti>=0.3.0')
+
+
+class KITTIBboxDataset(GetterDataset):
+
+    """Bounding box dataset for `KITTI dataset`_.
+
+    .. _`KITTI dataset`: http://www.cvlibs.net/datasets/kitti/raw_data.php
+
+    Args:
+        data_dir (string): Path to the root of the training data. If this is
+            :obj:`auto`, this class will automatically download data for you
+            under :obj:`$CHAINER_DATASET_ROOT/pfnet/chainercv/kitti`.
+        date ({'2011_09_26', '2011_09_28', '2011_09_29',
+                                           '2011_09_30', '2011_10_03'}):
+            reference Calibration datas.
+        drive_num ({'0xxx'}): get datas drive No.
+        sync (bool): get timer sync/nosync data.
+        is_left (bool): left/right camera image use 2type.
+        tracklet (bool): 3d bblox data. date only 2011_09_26.
+
+    This dataset returns the following data.
+
+    .. csv-table::
+        :header: name, shape, dtype, format
+
+        :obj:`img`, ":math:`(3, H, W)`", :obj:`float32`, \
+        "RGB, :math:`[0, 255]`"
+        :obj:`bbox` [#kitti_bbox_1]_, ":math:`(R, 4)`", :obj:`float32`, \
+        ":math:`(y_{min}, x_{min}, y_{max}, x_{max})`"
+        :obj:`label`, scalar, :obj:`int32`, ":math:`[0, \#class - 1]`"
+
+    .. [#kitti_bbox_1] If :obj:`tracklet = True`, \
+        :obj:`bbox` and :obj:`label` contain crowded instances.
+
+    When queried by an index, if :obj:`tracklet == True`,
+    this dataset returns a corresponding
+    :obj:`img, bbox, label`, a tuple of an image, bounding boxes, labels.
```

Comment:
Could you make two classes: `KITTIBboxDataset` and `KITTIImageDataset`?
It is confusing that when `tracklet=False`, the dataset never returns unempty bounding box or label even though it is named as `BboxDataset`.
`KITTIBboxDataset` should only support `2011_09_26`, and `KITTIImageDataset` should support all the other dates.
Also, if you don't want to, you do not need to implement `KITTIImageDataset`.

We have two classes for Cityscapes as well because subset of the dataset does not contain annotation.